src/ResourceManager/EventGrid/Commands.EventGrid.Test/ScenarioTests/TopicTests.ps1

```diff
@@ -226,3 +226,168 @@ function TopicNewKeyTests {
+    }
+    catch
+    {
+        Assert-True {$true}
+    }
+
+    try
+    {
+        Write-Debug "Creating a fifth EventGrid topic: $topicName5 in resource group $resourceGroupName with InputSchema $inputSchemaCustomEventSchema1"
+        $result = New-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -Name $topicName5 -Location $location -InputSchema $inputSchemaCustomEventSchema1
+        Assert-True {$false} "New-AzureRmEventGridTopic succeeded while it is expected to fail as inputmapping parameters are null"
+    }
+    catch
+    {
+        Assert-True {$true}
+    }
+
+    Write-Debug "Creating a fifth EventGrid topic: $topicName5 in resource group $resourceGroupName with InputSchema $inputSchemaCustomEventSchema1"
+    $result = New-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -Name $topicName5 -Location $location -InputSchema $inputSchemaCustomEventSchema1 -InputMappingField @{ id = "MyIdField"; topic = "MyTopicField"; eventtime = "MyEventTimeField"; subject = "MySubjectField"; eventtype = "MyEventTypeField"; dataversion = "MyDataVersionField" }
+    Assert-True {$result.ProvisioningState -eq "Succeeded"}
+    Assert-True {$result.InputSchema -eq $expectedInputMappingCustomEventSchema} "InputSchema is not correct. CustomEventSchema is expected"
+
+    Write-Debug "Creating a sixth EventGrid topic: $topicName6 in resource group $resourceGroupName with InputSchema $inputSchemaCustomEventSchema2"
+    $result = New-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -Name $topicName6 -Location $location -InputSchema $inputSchemaCustomEventSchema2 -InputMappingField @{ id = "MyIdField"; topic = "MyTopicField"; eventtime = "MyEventTimeField"; subject = "MySubjectField"; eventtype = "MyEventTypeField"; dataversion = "MyDataVersionField" } -InputMappingDefaultValue @{ subject = "MySubjectDefaultValue"; eventtype = "MyEventTypeDefaultValue"; dataversion = "MyDataVersionDefaultValue" }
+    Assert-True {$result.ProvisioningState -eq "Succeeded"}
+    Assert-True {$result.InputSchema -eq $expectedInputMappingCustomEventSchema} "InputSchema is not correct. CustomEventSchema is expected"
+
+    Write-Debug "Getting the created topic within the resource group"
+    $createdTopic = Get-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -Name $topicName5
+    Assert-True {$createdTopic.Count -eq 1}
+    Assert-True {$createdTopic.TopicName -eq $topicName5} "Topic created earlier is not found."
+    Assert-True {$createdTopic.InputSchema -eq $expectedInputMappingCustomEventSchema} "InputSchema is not correct. CustomEventSchema is expected"
+
+    Write-Debug " Deleting topic: $topicName"
+    Remove-AzureRmEventGridTopic -ResourceGroup $resourceGroupName -Name $topicName
+
+    Write-Debug " Deleting topic: $topicName2 using the ResourceID parameter set"
+    # Offline playback of tests is failing if I use Get-AzureRmResource, hence temporarily commenting this out
```

Comment:
Remove this comment - it is fine not to have this call in the test as long as you have validated it manually.

Reply:
Done